### ISSUE TITLE
Feature request: Plot a scalar field with geographical coordinates on map with projection 

### ISSUE THREAD

**Issue by steekhutzie** [plotly/plotly.py#715]
I have an array of global temperatures and associated longitude and latitude coordinates arrays. I would like to plot the temperatures as a heatmap on a map with custom projection and showing coastlines.
A working code example using matplotlib is:
```python
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

f, ax = plt.subplots(subplot_kw={'projection': ccrs.Robinson()})
p = ax.pcolormesh(longitude, latitude, temperature, cmap='RdBu_r', transform=ccrs.PlateCarree())
ax.coastlines()
f.colorbar(p, orientation='horizontal')
```
The output figure from code block above is (data available [here](https://www.esrl.noaa.gov/psd/data/gridded/data.noaa.oisst.v2.html)):
![map](https://cloud.githubusercontent.com/assets/9553668/24002863/1d11b742-0a62-11e7-866b-7c5ca2b49d7e.png)

At the moment I was only able to use plotly for a simple heatmap with the following code:
```python
from plotly.offline import iplot
import plotly.graph_objs as go

layout = go.Layout(geo={'projection': {'type': 'robinson'}})
data = go.Heatmap(x=longitude, y=latitude, z=temperature)

fig = go.Figure(data=[data], layout=layout)
iplot(fig)
```
As next figure shows, main issues are:

- the projection is not working;

- no coastlines are shown.
![heatmap](https://cloud.githubusercontent.com/assets/9553668/24004032/7fc01e30-0a65-11e7-96d9-c425204d43ea.png)

**Comment by Balinus** [comment]
Would also love something similar!

**Comment by empet** [comment]
You can plot both heatmap and contour plots on maps. Take a look here: https://plot.ly/~Dreamshot/9147 or visit this user (Dreamshot) Plotly profile for more examples.

**Comment by Balinus** [comment]
Thanks! Will take a look :)

**Comment by Balinus** [comment]
Although it does not seem to support mapping per se. I mean, the data are on a lat-lon coordinates, but what if we want to change the projection? I don't see a proper support for maps right out of the box (or at least without creating a unique map for each use case).

**Comment by JiaweiZhuang** [comment]
This blog should help: [How to Create 2D and 3D Interactive Weather Maps in Python and R](https://medium.com/@plotlygraphs/how-to-create-2d-and-3d-interactive-weather-maps-in-python-and-r-77ddd53cca8)

**Comment by jonmmease** [comment]
A geographic heatmap figure factory would be really nice!

**Comment by pedroaugustosmribeiro** [comment]
I also would like to plot any scallar (numpy array, xarray, etc) with a projeceted basemap in plotly just like is done on matplolib+cartopy integration

**Comment by Balinus** [comment]
I am currently using GeoMakie.jl (https://github.com/MakieOrg/GeoMakie.jl). Works really well!

**Comment by Balinus** [comment]
> > I am currently using GeoMakie.jl (https://github.com/MakieOrg/GeoMakie.jl). Works really well!
> 
> Nice, however it's in Julia. Is it possible to mix it inside a Python Plotly Dash app?

Not sure. Using Python inside Julia is easy, the other way around is more difficult in my experience. This is actually the reason why I use Julia : this gives me access to Julia and 100% of Python's ecosystem.

**Comment by g-voelker** [comment]
Of course, there are other methods out there, for Python cartopy stays the library of choice when it comes to map-based plots.  A speedy and interactive data visualization tool stays, however, a rare feat. 

I support the request for the feature and think it would make a good unique selling point for plotly in the geosciences and adjacent communities.
